### DISCUSSION TITLE
SSL HandShake error and Network unreachable while connecting to Upstream

### DISCUSSION THREAD

**Discussion by KnightKrusty** [minio/minio#12667] [Q&A]
```
server {
 listen 80;
 listen [::]:80;
 server_name s3.domain.com;
}
server {
  listen 443 ssl http2;
  listen [::]:443 ssl http2;
  server_name s3.domain.com;

  ssl_protocols TLSv1.2 TLSv1.3;
  ssl_ciphers HIGH:!MEDIUM:!LOW:!aNULL:!NULL:!SHA;
  ssl_prefer_server_ciphers on;
  ssl_session_cache shared:SSL:10m;

  keepalive_timeout    70;
  sendfile             on;
  client_max_body_size 80m;


  ssl_certificate     /etc/nginx/ssl/cert.pem;
  ssl_certificate_key /etc/nginx/ssl/cert.key.pem;


 # To allow special characters in headers
 ignore_invalid_headers off;
 # Allow any size file to be uploaded.
 # Set to a value such as 1000m; to restrict file size to a specific value
 # client_max_body_size 0;
 # To disable buffering
 proxy_buffering off;



 location / {
   proxy_set_header X-Real-IP $remote_addr;
   proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
   proxy_set_header X-Forwarded-Proto $scheme;
   proxy_set_header Host $http_host;

   proxy_connect_timeout 300;
# Default is HTTP/1, keepalive is only enabled in HTTP/1.1
   proxy_http_version 1.1;
   proxy_set_header Connection "";
   chunked_transfer_encoding off;

   proxy_pass http://127.0.0.1:9000;
   # If you are using docker-compose this would be the hostname i.e. minio
   # Health Check endpoint might go here. See https://www.nginx.com/resources/wiki/modules/h$
   # /minio/health/live;
 }
}


```

This is my nginx and whenever i try to cache the bucket using nginx reverse proxy i get these error.

```
upstream: "https://104.21.235.34:443/media_attachments/files/011/910/139/original/a56ed76cdc40e2a6.jpeg"

SSL Handshake Errors
SSL_do_handshake() failed (SSL: error:14094410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure:SSL alert number 40) while SSL handshaking to upstream, client

In some cases it resolves to IPV6
 upstream: "https://[2606:4700:3038::6815:eb21]:443/media_attachments/files/011/910/139/original/a56ed76cdc40e2a6.jpeg",


```

**Comment by harshavardhana** [comment]
> upstream: "https://104.21.235.34:443/media_attachments/files/011/910/139/original/a56ed76cdc40e2a6.jpeg"
> 
> SSL Handshake Errors
> SSL_do_handshake() failed (SSL: error:14094410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure:SSL alert number 40) while SSL handshaking to upstream, client
> 
> In some cases it resolves to IPV6
>  upstream: "https://[2606:4700:3038::6815:eb21]:443/media_attachments/files/011/910/139/original/a56ed76cdc40e2a6.jpeg",

how is this MinIO issue? @KnightKrusty

**Reply by KnightKrusty** [reply]
Everytime Nginx tries to access the media files in minio it simply throw these error so i wanted to ask is there anything wrong with nginx config in minio(the one i attached here).

I did managae to solve ssl hand shake issue (i guess doesnt see it in the log anymore).

The Main problem is this  no live upstreams while connecting to upstream. Reading on the internet it seems the issue is the backend not nginx itself, the server cant respond fast enough. can you provide any info over that?

**Reply by harshavardhana** [reply]
> The Main problem is this no live upstreams while connecting to upstream. Reading on the internet it seems the issue is the backend not nginx itself, the server cant respond fast enough. can you provide any info over that?

If you have an actual bug with a reproducer please report it we will fix it otherwise I don't know what else I can help here. 

Performance issues and architecture guidance we only investigate for customers that have subscription with us, if this a production use case look at https://min.io/pricing .

**Reply by KnightKrusty** [reply]
My setup runs on single powerfull VPS with dedicated cores and disk attached to it in LVM. I created this to test in my application and use it for development puposes and learning about disk management in Linux. if i would have to buy your plans it would be much cheaper to just buy a bucekt from wasabi, spaces, or vultr. Good Pricing and will be alot more cheaper.

I dont have any problem with your bussiness model but telling everyone even small user to just buy your support plan is not a very good tactics. 

Will be switching to spaces.
